### DISCUSSION TITLE
Toolbar item  to add links to Heading or  link block  inside the note

### DISCUSSION THREAD

**Discussion by VigneshR387** [chrisgurney/obsidian-note-toolbar#205] [Q&A] [answered]
A toolbar item  when clicked will navigate to a specific  link block or Heading inside my note.The  Only Current workaround for this is adding heading links inside properties and then adding properties as my toolbar item  .

**Comment by chrisgurney** [comment, marked as answer]
@VigneshR387 There's a couple existing options here. Let me know if this helps!

## URI item

Create a `URI` item and set the link to `#My Heading`. Clicking on the item will jump to the heading and highlight it, the same as creating an [internal link](https://help.obsidian.md/Linking+notes+and+files/Internal+links#Link%20to%20a%20heading%20in%20a%20note).

[Linking to a block](https://help.obsidian.md/Linking+notes+and+files/Internal+links#Link+to+a+block+in+a+note) should work from `URI` items as well, though I haven't tried it.

## Templater script

Here's another solution that uses a Templater script, but just jumps the cursor to the heading (you could easily modify this to go to the line after the heading if you wish).

> This doesn't use anything specific to Templater, so it could be translated with minimal effort to work with Dataview or JS Engine.

Just save the below to a JavaScript file, and see the directions below.

LMK if you try this out, or need a hand getting it to work. (I have another version that shows a selector, that lets you choose a heading to jump to; LMK if you're interested in that.)

For linking to blocks, I'm guessing there's a similar solution (possibly via the `sections` object in the script below?), but will need some research -- or direction from another user -- to figure it out. 

```javascript
/**
 * Navigates to the heading provided in the current note.
 * 
 * Usage:
 * - In the Templater plugin's settings, add the folder this script is in under "User script functions".
 * - Use tp.user.GoToHeading in "Execute Templater command".
 * 
 * Example:
 * tp.user.GoToHeading(tp, "My Heading")
 */

module.exports = async function goToHeading(tp, headingToGoTo) {
	const currentFile = app.workspace.getActiveFile();
	const { headings, sections } = app.metadataCache.getFileCache(currentFile);
	
	if (headings) {
		const selectedHeading = headings.filter(heading => heading.heading === headingToGoTo);
		const headingLine = selectedHeading[0].position.start.line;
		
		const editor = app.workspace.activeLeaf.view?.editor;
		editor.setSelection({line: headingLine, ch: 0});
	}
}
```

**Reply by VigneshR387** [reply]
Thanks for the help
